File: scripts/packageZip.js
import fs from "fs";
import path from "path";
import { fileURLToPath } from "url";
import { createRequire } from "module";

const require = createRequire(import.meta.url);
const archiver = require("archiver");

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
const rootDir = path.resolve(__dirname, "..");
const workspaceRoot = path.resolve(rootDir, "..");

const zipPaths = [
  path.join(rootDir, "Growly_MERN_Fullstack.zip"),
  path.join(workspaceRoot, "Growly_MERN_Fullstack.zip"),
];

async function createZip(outPath) {
  return new Promise((resolve, reject) => {
    const output = fs.createWriteStream(outPath);
    const archive = archiver("zip", { zlib: { level: 9 } });

    output.on("close", () => {
      console.log(
        `📦 Created ${path.basename(outPath)} (${(archive.pointer() / 1024).toFixed(1)} KB) at: ${outPath}`
      );
      resolve();
    });

    archive.on("error", (err) => reject(err));
    archive.pipe(output);

    archive.glob("**/*", {
      cwd: rootDir,
      ignore: [
        "node_modules/**",
        "dist/**",
        ".git/**",
        "*.zip",
        ".system_generated/**",
        "scratch/**",
      ],
      dot: true,
    });

    archive.finalize();
  });
}

async function run() {
  for (const p of zipPaths) {
    await createZip(p);
  }
}

run().catch((e) => {
  console.error("Packaging error:", e);
  process.exit(1);
});

File: scripts/package_zip.py
import os
import zipfile

def create_zip(source_dir, output_zip):
    print(f"Creating zip: {output_zip} from {source_dir}...")
    exclude_dirs = {"node_modules", "dist", ".git", ".system_generated", "scratch"}
    exclude_extensions = {".zip"}

    with zipfile.ZipFile(output_zip, "w", zipfile.ZIP_DEFLATED) as zipf:
        for root, dirs, files in os.walk(source_dir):
            dirs[:] = [d for d in dirs if d not in exclude_dirs]

            for file in files:
                ext = os.path.splitext(file)[1].lower()
                if ext in exclude_extensions:
                    continue

                full_path = os.path.join(root, file)
                rel_path = os.path.relpath(full_path, source_dir)
                zipf.write(full_path, rel_path)

    size_kb = os.path.getsize(output_zip) / 1024
    print(f"[OK] Created {output_zip} ({size_kb:.1f} KB)")

if __name__ == "__main__":
    base_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
    workspace_root = os.path.dirname(base_dir)

    target_1 = os.path.join(base_dir, "Startup_IQ_Fullstack.zip")
    target_2 = os.path.join(workspace_root, "Startup_IQ_Fullstack.zip")

    create_zip(base_dir, target_1)
    create_zip(base_dir, target_2)
